components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2783,4 +2820,103 @@ void RewardsServiceImpl::OnDeleteActivityInfo(
   }
 }
 
+void RewardsServiceImpl::GetAutoContributeCount(
+    const rewards_counter::OnRewardsCountedCallback callback) {
+  GetReconcileStamp(
+      base::Bind(&RewardsServiceImpl::GetCountByReconcileStamp,
+        AsWeakPtr(), callback));
+}
+
+void RewardsServiceImpl::GetLastReconcileStamp(
+    const rewards_counter::OnRewardsCountedCallback callback,
+    int64_t count) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&AutoContributionLastReconcileDateOnFileTaskRunner,
```

Comment:
why are we making yet another db call here? You have at least 3 separate db calls that could be combined into a single query, but you also keep needlessly posting back and forth to the UI thread. Once you're on the task runner thread you can make as many DB calls as you want without going back and forth

Reply:
just like you're doing in `RemoveDataOnFileTaskRunner`

Reply:
The problem here is I have to fetch 2 different values to send back. The count and separately, the previous reconcile date, both to be used separately.

Reply:
Addressed in DM and resolved.